src/file_analysis/AnalyzerSet.cc

```diff
@@ -20,7 +20,7 @@ static void analyzer_del_func(void* v)
 	delete a;
 	}
 
-AnalyzerSet::AnalyzerSet(File* arg_file) : file(arg_file)
+AnalyzerSet::AnalyzerSet(File* arg_file) : file(arg_file), analyzer_map(zeek::DictOrder::ORDERED)
```

Comment:
@timwoj - this would probably collide with your `std::map` work. As in the PR description, I think there's value in having the order of analyzer be predictable.

Reply:
And even more: A `std::map` wouldn't be ordered by insert order which is leveraged here.

I'm leaving this open as I'm thinking it's net positive. On the one hand it's the analyzers field order in the log that becomes stable, but on the other it's also having a defined/predictable order in which file analyzers are processed. That should make it more attainable to debug or reproduce issues that rely on the order of file analyzers being processed.

Reply:
The thing about "ordered" mode in our existing dictionary is that it's implemented by just keeping a vector with the insertion order. When you iterate over the dictionary, it instead iterates over that vector and does lookups against the dictionary. We could do the same for any implementation using one of the std maps.

Reply:
Not sure I like this change. Couple things: (1) it could be considered a feature, not a bug, to have an unstable order: allowing the order to change may reveal bugs that happen to be invisible in particular order; and (2) I'm sure we have plenty other places where we could make the same argument, and changing it just here but not elsewhere feels inconsistent and potentially confusing to me.